game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameSelectorModel.java

```diff
@@ -186,10 +189,28 @@ public void loadDefaultGameSameThread() {
         // can load. (ie: if a previous version of triplea was using running a game within its
         // root folder, we shouldn't open it)
         .filter(
-            defaultGame ->
-                defaultGame.contains(ClientFileSystemHelper.getUserRootFolder().toURI().toString()))
-        .filter(defaultGame -> new File(defaultGame).exists())
+            defaultGame -> {
+              String rootFolder = ClientFileSystemHelper.getUserRootFolder().toURI().toString();
```

Comment:
I wonder if the check for a given parent folder would not be easier to use the `Path` API and  do a recursive check of parents or use the `iterator()` method to check folders. The special casing by OS is hard to grok, I could envision someone testing this and thinking it is unnecessary code (also we really want the code to be as cross-platform as possible, if this needs modification, most of us have no access to a MAC and would SOL).

We have a `FileUtils` class, as a suggestion, what do you think of creating a helper method there to check if a given parent folder exists on a given path, that can be tested readily, and then use that utility method here?

Reply:
`Path#startsWith` does the checking so I switched to that once I've created a Path object.